### PR TITLE
Clarify formatting in dune

### LINKED ISSUE #3854: unclear documentation of automatting formatting support in dune jang 2.0
I was confused by this line from [the docs][]

[the docs]: https://dune.readthedocs.io/en/stable/formatting.html?highlight=automatic%20formatting#configuring-automatic-formatting-dune-2-0

```
If using (lang dune 2.0), there is nothing to do, formatting will be set up by default.
```

Would it be possible to revise it to mention that one must still create a `.ocamlformat` file?

## Expected Behavior

I interpreted the line to mean,

1. Specify `(lang dune 2.0)` in `dune-project`
2. Run `dune build @fmt --auto-promote`
3. See this file
   
    ```
    let
    ()
    =
    ()
    ```
   
    become this file:
   
    ```
    let () = ()
    ```

## Actual Behavior

In fact, it turns out dune automatically tries to run ocamlformat, but I still need to set up ocamlformat in my repo by creating a `.ocamlformat` file at the repo root.  An empty file is fine.

## Reproduction

See `Expected behavior`

## Specifications

See link to docs at top.

## Additional information 

Not a code bug.

### PR DESCRIPTION
**Pull request by rgrinberg**
Fix #3854

### CODE REVIEW

File: doc/formatting.rst [hunk 1]
```diff
@@ -18,8 +18,9 @@ Furthermore it can be used to format code of any defined dialect (see
 Configuring automatic formatting (dune 2.0)
 ===========================================
 
-If using ``(lang dune 2.0)``, there is nothing to do, formatting will be set up
-by default.
+If using ``(lang dune 2.0)``, there is nothing to setup in dune, formatting will
+be set up by default. However, ocamlformat will still refuse to format sources
+without an .ocamlformat file present in the project root.
```
Comment (nojb) [inline doc/formatting.rst:23]:
```suggestion
without an ``.ocamlformat`` file present in the project root.
```

File: doc/formatting.rst [hunk 2]
```diff
@@ -18,8 +18,9 @@ Furthermore it can be used to format code of any defined dialect (see
 Configuring automatic formatting (dune 2.0)
 ===========================================
 
-If using ``(lang dune 2.0)``, there is nothing to do, formatting will be set up
-by default.
+If using ``(lang dune 2.0)``, there is nothing to setup in dune, formatting will
+be set up by default. However, ocamlformat will still refuse to format sources
```
Comment (nojb) [inline doc/formatting.rst:22]:
```suggestion
be set up by default. However, ``ocamlformat`` will still refuse to format sources
```

**Review by nojb — APPROVED** [review]
LGTM

File: doc/formatting.rst (hunk 1 above)
Comment (rgrinberg) [reply doc/formatting.rst]: Thanks

File: doc/formatting.rst (hunk 2 above)
Comment (rgrinberg) [reply doc/formatting.rst]: Actually, I think that it would be `ocamlformat_`. It's a reference, rather than code.

Comment (nojb) [reply doc/formatting.rst]: Indeed, thanks.

**Pull request merged**